core/src/test/scala/unit/kafka/server/ServerGenerateClusterIdTest.scala

```diff
@@ -0,0 +1,113 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package kafka.server
+
+import java.util.Properties
+
+import kafka.server.{KafkaConfig, KafkaServer}
+import kafka.utils.{CoreUtils, TestUtils, ZkUtils}
+import kafka.zk.ZooKeeperTestHarness
+import org.junit.Assert._
+import org.junit.{Before, Test}
+
+class ServerGenerateClusterIdTest extends ZooKeeperTestHarness {
+  var props1: Properties = null
+  var config1: KafkaConfig = null
+  var props2: Properties = null
+  var config2: KafkaConfig = null
+  var props3: Properties = null
+  var config3: KafkaConfig = null
+
+  @Before
+  override def setUp() {
+    super.setUp()
+    props1 = TestUtils.createBrokerConfig(1, zkConnect)
+    config1 = KafkaConfig.fromProps(props1)
+    props2 = TestUtils.createBrokerConfig(2, zkConnect)
+    config2 = KafkaConfig.fromProps(props2)
+    props3 = TestUtils.createBrokerConfig(3, zkConnect)
+    config3 = KafkaConfig.fromProps(props3)
+  }
+
+  @Test
+  def testAutoGenerateClusterId() {
+    //Make sure that the cluster id doesn't exist yet.
+    assertFalse(zkUtils.pathExists(ZkUtils.ClusterIdPath))
+    var server1 = new KafkaServer(config1, threadNamePrefix = Option(this.getClass.getName))
+    server1.startup()
+
+    // Make sure the cluster id is 48 characters long (base64 of UUID.randomUUID() )
+    val clusterIdOnFirstBoot = server1.clusterId
+    assertEquals(clusterIdOnFirstBoot.length, 48)
+
+    server1.shutdown()
+
+    //Make sure that the cluster id is persistent.
+    assertTrue(zkUtils.pathExists(ZkUtils.ClusterIdPath))
+    assertEquals(zkUtils.getClusterId().get, clusterIdOnFirstBoot)
```

Comment:
Avoid the `get` and add `Some` to the right value. That way you get a more useful error message if the test fails.